# bioinformatics
Several projects and works from the bioinformatics course   
* global_aligners: <br>Programs calculate the optimal global alignment of two sequences using the Needleman-Wunsch algorithm, using a DNAfull penalty matrix and an affine or linear gap penalty.<br>
* mapper: <br>The mapper based on the construction of a hash table for the reference genome. The program accepts a genome in FASTA format, the number k, creates a hash table and saves it to a file. After that, it reads this file and reads in FASTQ format and performs mapping of the reads (without gaps).<br>
*(a file with a hash table weighs a lot, so a part of it is attached as an example)*<br>
* human_dna_project: <br>The program determines at what minimum number of k there is a nucleotide sequence of length k that does not occur in the human genome and counts the number of such sequences.
* covid_project: <br>1. Comparison of SARS-CoV-2 and SARS-CoV coronaviruses by amino acid sequences of NSP12 protein Spike proteins.<br>2. Estimation of the average number of mutations in Spike and NCP12 proteins in SARS-CoV-2 coronavirus sequences (Russian samples) relative to the Wuhan reference.<br>3. The most frequent single nucleotide substitution in a Spike protein.<br>4. Finding the most evolutionarily close and most distant viruses (among SARS-CoV-2, SARS-CoV, MERS-CoV, HCoV-OC43, HCoV-NL63, HCoV-HKU1, HCoV-229E), based on the score of alignment of their genome sequences.<br>5. Creating a hash table for the k-mers of the SARS-CoV-2 genome. Searching for the minimum value of k, at which exactly one position corresponds to each fragment. <br>
